plotly/figure_factory/_bullet.py

```diff
@@ -0,0 +1,257 @@
+                hoverinfo='x' if as_rows else 'y',
+                orientation='h' if as_rows else 'v',
+                width=0.4,
+                base=0,
+                xaxis='x{}'.format(row + 1),
+                yaxis='y{}'.format(row + 1)
+            )
+            fig['data'].append(bar)
+
+        # markers
+        x = df.iloc[row]['markers'] if as_rows else [0.5]
+        y = [0.5] if as_rows else df.iloc[row]['markers']
+        markers = go.Scatter(
+            x=x,
+            y=y,
+            marker=dict(
+                color='rgb(0, 0, 0)',
+                symbol=marker_symbol,
+                size=marker_size
+            ),
+            name='markers',
+            hoverinfo='x' if as_rows else 'y',
+            xaxis='x{}'.format(row + 1),
+            yaxis='y{}'.format(row + 1)
+        )
+        fig['data'].append(markers)
+
+        # labels
+        title = df.iloc[row]['title']
+        if 'subtitle' in df:
+            subtitle = '<br>{}'.format(df.iloc[row]['subtitle'])
+        else:
+            subtitle = ''
+        label = '<b>{}</b>'.format(title) + subtitle
+        annot = utils.annotation_dict_for_label(
+            label,
+            (num_of_lanes - row if as_rows else row + 1),
+            num_of_lanes, subplot_spacing,
+            'row' if as_rows else 'col',
+            True if as_rows else False,
+            False
+        )
+        fig['layout']['annotations'].append(annot)
+
+    return fig
+
+
+def create_bullet(df, as_rows=True, marker_size=16,
+                  marker_symbol='diamond-tall', range_colors=None,
+                  measure_colors=None, subplot_spacing=None,
+                  title='Bullet Chart', height=600, width=1000):
+    """
+    Returns figure for bullet chart.
+
+    :param (pd.DataFrame | list) df: either a JSON list of dicts or a pandas
+        DataFrame. All keys must be one of 'title', 'subtitle', 'ranges',
+        'measures', and 'markers'.
+    :param (bool) as_rows: if True, the bars are placed horizontally as rows.
+        If False, the bars are placed vertically in the chart.
+    :param (int) marker_size: sets the size of the markers in the chart.
+    :param (str | int) marker_symbol: the symbol of the markers in the chart.
+        Default='diamond-tall'
+    :param (list) range_colors: a list of two colors between which all
+        the rectangles for the range are drawn. These rectangles are meant to
+        be qualitative indicators against which the marker and measure bars
+        are compared.
+        Default=['rgb(198, 198, 198)', 'rgb(248, 248, 248)']
+    :param (list) measure_colors: a list of two colors which is used to color
+        the thin quantitative bars in the bullet chart.
+        Default=['rgb(31, 119, 180)', 'rgb(176, 196, 221)']
+    :param (float) subplot_spacing: set the distance between each bar chart.
+        If not specified an automatic spacing is assigned based on the number
+        of bars to be plotted.
```

Comment:
What units are these in and what's the range of these numbers?

Reply:
0 to 1.